src/mesh/tools.rs

```diff
@@ -625,54 +622,45 @@ mod tests {
-            v(-1.0, 1.0, 1.0, position, scale),
-            v(1.0, 1.0, 1.0, position, scale),
-            v(1.0, 1.0, -1.0, position, scale),
-            // front
-            v(-1.0, -1.0, -1.0, position, scale),
-            v(1.0, -1.0, -1.0, position, scale),
-            v(1.0, -1.0, 1.0, position, scale),
-            v(-1.0, -1.0, 1.0, position, scale),
+    pub fn welded_box_smooth_mesh() -> Mesh {
+        let vertices = vec![
+            Point3::new(1.0, 1.0, 1.0),
+            Point3::new(-1.0, -1.0, 1.0),
+            Point3::new(-1.0, 1.0, 1.0),
+            Point3::new(1.0, 1.0, -1.0),
+            Point3::new(-1.0, 1.0, -1.0),
+            Point3::new(-1.0, -1.0, -1.0),
+            Point3::new(1.0, -1.0, 1.0),
+            Point3::new(1.0, -1.0, -1.0),
         ];
 
-        let vertex_normals = vec![
-            n(-1.0 / 3.0, 1.0 / 3.0, -1.0 / 3.0),
-            n(-1.0 / 3.0, 1.0 / 3.0, 1.0 / 3.0),
-            n(1.0 / 3.0, 1.0 / 3.0, 1.0 / 3.0),
-            n(1.0 / 3.0, 1.0 / 3.0, -1.0 / 3.0),
-            // front
-            n(-1.0 / 3.0, -1.0 / 3.0, -1.0 / 3.0),
-            n(1.0 / 3.0, -1.0 / 3.0, -1.0 / 3.0),
-            n(1.0 / 3.0, -1.0 / 3.0, 1.0 / 3.0),
-            n(-1.0 / 3.0, -1.0 / 3.0, 1.0 / 3.0),
+        let normals = vec![
+            n(0.25, 0.5, 0.25),
+            n(-0.5, -0.25, 0.25),
+            n(-0.25, 0.25, 0.5),
+            n(0.5, 0.25, -0.25),
+            n(-0.33333334, 0.33333334, -0.33333334),
+            n(-0.25, -0.5, -0.25),
+            n(0.33333334, -0.33333334, 0.33333334),
+            n(0.25, -0.25, -0.5),
         ];
 
         let faces = vec![
-            // back
-            TriangleFace::new(0, 1, 2),
-            TriangleFace::new(2, 3, 0),
-            // front
-            TriangleFace::new(4, 5, 6),
-            TriangleFace::new(6, 7, 4),
-            // top
-            TriangleFace::new(7, 6, 1),
-            TriangleFace::new(2, 1, 6),
-            // bottom
-            TriangleFace::new(5, 0, 3),
-            TriangleFace::new(0, 5, 4),
-            // right
-            TriangleFace::new(6, 3, 2),
-            TriangleFace::new(3, 6, 5),
-            // left
-            TriangleFace::new(4, 7, 0),
-            TriangleFace::new(1, 0, 7),
+            TriangleFace::new_separate(0, 4, 2, 0, 4, 2),
+            TriangleFace::new_separate(0, 3, 4, 0, 3, 4),
+            TriangleFace::new_separate(5, 7, 6, 5, 7, 6),
+            TriangleFace::new_separate(1, 5, 6, 1, 5, 6),
+            TriangleFace::new_separate(1, 6, 2, 1, 6, 2),
+            TriangleFace::new_separate(0, 2, 6, 0, 2, 6),
+            TriangleFace::new_separate(3, 7, 4, 3, 7, 4),
+            TriangleFace::new_separate(4, 7, 5, 4, 7, 5),
+            TriangleFace::new_separate(0, 6, 3, 0, 6, 3),
+            TriangleFace::new_separate(3, 6, 7, 3, 6, 7),
+            TriangleFace::new_separate(1, 4, 5, 1, 4, 5),
+            TriangleFace::new_separate(1, 2, 4, 1, 2, 4),
```

Comment:
Since indices for vertices and normals are exactly the same, you can use `TriangleFace::new` here. I see you always pick `new_separate` even if not necessary. Is that just habit or do you have a reason for that? We could remove `new` and rename `new_separate` to `new`.

Reply:
I don't know. It just seems more correct. The case for current `new` seems to be very specific and is a subset of the `new_separate`. If we should make a change, I suggest keeping both constructors but making the explicit one default?
And yes, I should have used the other one here.

Reply:
I am ok also with what you are suggesting. I would also be ok with just one constructor. Up to you.